cache/src/main/kotlin/com/dropbox/android/external/cache4/RealCache.kt

```diff
@@ -139,12 +148,22 @@ internal class RealCache<Key : Any, Value : Any>(
         val existingEntry = cacheEntries[key]
         if (existingEntry != null) {
             // cache entry found
-            recordWrite(existingEntry, nowNanos)
+            // remove the weight of the current entry
+            totalWeight -= existingEntry.weight
```

Comment:
The cache is multi-threaded and this is not thread safe. 

This is actually not easy to fix without adding substantial synchronization to the cache (or adding other optimizations beyond the scope of this cache). The reason is that you need to make sure the state of `cachedEntries` and `totalWeight` is updated atomically.